### PR TITLE
testdata: get test assets related to package root instead of GOPATH

### LINKED ISSUE #2270: Tests fail when run as part of a module build
### What version of gRPC are you using?

v1.14.0

### What version of Go are you using (`go version`)?

go version go1.11rc1 linux/amd64

### What operating system (Linux, Windows, …) and version?

Fedora 28

### What did you do?

A project I develop imports gRPC as a [Go module](https://github.com/golang/go/wiki/Modules).  When tests are run as part of `go test all`, tests for the `google.golang.org/grpc` and `google.golang.org/grpc/credentials` packages fail due to being unable to look up the testdata directory relative to one's GOPATH.  In a module build, GOPATH is not taken into consideration, so any relative paths to the testdata which depend on GOPATH cannot work.

It is still possible to use relative paths to testdata in a module build (e.g. paths beginning with '.' or '..'), as long as the referenced path is also part of the same module.  It is not possible to reference testdata outside of the same module being tested, as these files are not available to the module and are not included in the module archive (a zip file).

Since gRPC has not converted to modules (there is no go.mod file), when used in a module build, the entire repository is used as a single module.  This means that relative paths to testdata are possible as long as they reference files in the same repository.  However, keep in mind that if the repo is split into various submodules in the future, the relative testdata directories may once again point to a different module's files and be inaccessible.

### What did you expect to see?

Passing tests

### What did you see instead?

```
2018/08/19 13:13:20 Error finding google.golang.org/grpc/testdata directory: file does not exist
FAIL	google.golang.org/grpc	13.919s
2018/08/19 13:13:06 Error finding google.golang.org/grpc/testdata directory: file does not exist
FAIL	google.golang.org/grpc/credentials	0.049s
```

### PR DESCRIPTION
**Pull request by menghanl**
fixes #2270

### CODE REVIEW

**Comment by menghanl** [comment]
@jrick Can you try if #2270 is fixed by this? Thanks!

**Comment by jrick** [comment]
Yep, this fixes them.

**Comment by menghanl** [comment]
@jrick Thanks a lot for verifying!

File: testdata/testdata.go [hunk 1]
```diff
@@ -18,9 +18,14 @@
...
 
 import (
-	"log"
-	"os"
 	"path/filepath"
+	"runtime"
+)
+
+var (
+	_, b, _, _ = runtime.Caller(0)
```
Comment (dfawley) [inline testdata/testdata.go]: It doesn't really matter in such a small package, but in general: I don't like global variables that are a single letter.  Also it's only an intermediate involved in setting another global.  We could do something like this instead to hide it:

```go
func caller() string { _, b, _, _ = runtime.Caller(0); return b }

var basepath = filepath.Dir(caller())
```

or

```go
var basepath string
func init() { // set basepath }
```

Comment (menghanl) [reply testdata/testdata.go]: Fixed. Doing `init()`

**Pull request merged**